### PR TITLE
markdown text area

### LINKED ISSUE #12: Add a markdown parser to the site
We will be uploading markdown files to our site. So we need a parser to read and present the files

### PR DESCRIPTION
**Pull request by Maher9596**
I created a markdown text area and installed a parser **showdown** ,

I recreated this PR because the first time it was done i included unwanted files.

This PR should be solving issue #26.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,29 +1,17 @@
 <!DOCTYPE html>
```
Comment (Ma7eer) [inline index.html:1]: Why did you not create another html file? We still need the login file. You should create another html file and add the markdown to that.

File: index.html [hunk 2]
```diff
@@ -1,29 +1,17 @@
...
 <html>
     <head>
-        <title>Checklist</title>
-        <link rel="stylesheet" href="main.css">
+		<title>MarkDownTextArea</title>
```
Comment (Ma7eer) [inline index.html]:
```suggestion
		<title>MarkDown Text Area</title>
```

File: index.html [hunk 3]
```diff
@@ -1,29 +1,17 @@
...
+		<link rel="stylesheet" href="main.css">
+        <link rel="stylesheet" type="text/css" href="https://yarnpkg.com/en/package/normalize.css">
```
Comment (Ma7eer) [inline index.html:6]: Why are we using this library?

File: main.css [hunk 4]
```diff
@@ -1,3 +1,31 @@
-h1 {
-    color: red;
+body {
+    font-size: 18px;
```
Comment (Ma7eer) [inline main.css:2]: No **magic numbers** in our code.

File: main.css [hunk 5]
```diff
@@ -1,3 +1,31 @@
...
+}
+
+.main-content {
+    display: flex;
+    height: 50%;
+}
+
+.text-editor {
+    flex: 1;
+    padding:  30px ;
```
Comment (Ma7eer) [inline main.css:19]: No **magic numbers** in our code.

File: main.css [hunk 6]
```diff
@@ -1,3 +1,31 @@
...
+    background: #ffffff;
+    color: #333333;
+}
+
+.preview {
+    flex: 1;
+    padding: 0 50px;
+    background: #f3f3f3;
+    color: #3d3d3d;
 }
```
Comment (Ma7eer) [inline main.css:31]: In each file, make sure you have an extra line at the bottom.

File: main.js [hunk 7]
```diff
@@ -1 +1,14 @@
-console.log("HELLO")
\ No newline at end of file
+const textEditor = document.querySelector(".text-editor")
+const preview = document.querySelector(".preview")
+var showdown = require('showdown')
```
Comment (Ma7eer) [inline main.js:3]: Why `var` here instead of `const`?

**Review by Ma7eer — CHANGES_REQUESTED** [review]
Still some things to be worked on.

File: main.js (hunk 7 above)
Comment (Maher9596) [reply main.js:3]: There is no technical reason why I used **var** in this case as it's being defined globally, I should have used **const** for the sake of consistency.
The mistake i have done was to copy the code from the Package documentation.

File: main.css (hunk 6 above)
Comment (Maher9596) [reply main.css:31]: Noted.

File: main.css (hunk 5 above)
Comment (Maher9596) [reply main.css:19]: I understand the meaning of magic numbers, However i don't understand why r we calling these magic numbers.

File: main.css (hunk 4 above)
Comment (Maher9596) [reply main.css:2]: I understand the meaning of magic numbers, However i don't understand why r we calling these magic numbers.

File: index.html (hunk 3 above)
Comment (Maher9596) [reply index.html:6]: I used this library to test the styling it was supposed to be removed before pushing.

File: index.html (hunk 1 above)
Comment (Maher9596) [reply index.html:1]: I thought I can just replace the content of the pulled html file.

I will close this PR and I will create another one to make everything clear

**Pull request closed without merging**